posthog-core/src/index.ts

```diff
@@ -63,38 +57,422 @@ export abstract class PostHogCore {
     this.flushAt = options?.flushAt ? Math.max(options?.flushAt, 1) : 20
     this.flushInterval = options?.flushInterval ?? 10000
     this.captureMode = options?.captureMode || 'form'
-    this.sendFeatureFlagEvent = options?.sendFeatureFlagEvent ?? true
-    // If enable is explicitly set to false we override the optout
-    this._optoutOverride = options?.enable === false
+
     this._retryOptions = {
       retryCount: options?.fetchRetryCount ?? 3,
       retryDelay: options?.fetchRetryDelay ?? 3000,
     }
     this.requestTimeout = options?.requestTimeout ?? 10000 // 10 seconds
-    this._sessionExpirationTimeSeconds = options?.sessionExpirationTimeSeconds ?? 1800 // 30 minutes
+  }
 
-    // NOTE: It is important we don't initiate anything in the constructor as some async IO may still be underway on the parent
-    if (options?.preloadFeatureFlags !== false) {
-      safeSetTimeout(() => {
-        this.reloadFeatureFlags()
-      }, 1)
+  protected getCommonEventProperties(): any {
+    return {
+      $lib: this.getLibraryId(),
+      $lib_version: this.getLibraryVersion(),
     }
   }
 
-  protected getCommonEventProperties(): any {
-    const featureFlags = this.getFeatureFlags()
+  on(event: string, cb: (...args: any[]) => void): () => void {
+    return this._events.on(event, cb)
+  }
+
+  debug(enabled: boolean = true): void {
+    this.removeDebugCallback?.()
+
+    if (enabled) {
+      this.removeDebugCallback = this.on('*', (event, payload) => console.log('PostHog Debug', event, payload))
+    }
+  }
+
+  private buildPayload(payload: { distinct_id: string; event: string; properties?: PostHogEventProperties }): any {
+    return {
+      distinct_id: payload.distinct_id,
+      event: payload.event,
+      properties: {
+        ...(payload.properties || {}),
+        ...this.getCommonEventProperties(), // Common PH props
+      },
+    }
+  }
+
+  /***
+   *** TRACKING
+   ***/
+  identifyStateless(distinctId: string, properties?: PostHogEventProperties, options?: PosthogCaptureOptions): this {
+    // This identify doesn't add anything to set, you have to do it yourself!
+    const payload = {
+      ...this.buildPayload({
+        distinct_id: distinctId,
+        event: '$identify',
+        properties,
+      }),
+    }
+
+    this.enqueue('identify', payload, options)
+    return this
+  }
+
+  captureStateless(
```

Comment:
Wondering, should we make these methods `protected`?

When you do the node version, we wouldn't want these methods to be public right? So the only reason to not have them protected is if we want the stateful clients to be able to call the stateless method which I feel like we wouldn't want, or at least the naming wouldn't make sense in that context.

Reply:
the point is this is the purest form of the fn, so node and core would both call this!

See how node will work around this: https://github.com/PostHog/posthog-js-lite/pull/75

Reply:
Ah I misundestood the plan here perhaps... I figured the Node library would `extend` the stateless class and then call the methods that way, meaning they can remain `protected` as we wouldn't need two classes?

Maybe I need to have a deeper look at the other PR...

Reply:
Could do it like that, tbh. Yeah, makes sense! Will make these protected in that PR though, if I go down that route!

Reply:
Made a PR suggestion to validate my thinking - take it or leave it :D https://github.com/PostHog/posthog-js-lite/pull/76

Reply:
took it!